pydelighted/Delighted.py

```diff
@@ -1,3 +1,54 @@
+import os
+import requests
+import yaml
+
+from pydelighted.useful import process_data, flatten, get_column_names, send_temp_data, _clean
+
+
 class Delighted:
-    def __init__(self, dbstream):
-        self.dbstream=dbstream
\ No newline at end of file
+    def __init__(self, dbstream, var_env_key, config_file_path='config.yaml', version='v1'):
+        self.dbstream = dbstream
+        self.var_env_key = var_env_key
+        self.url = "https://api.delighted.com/%s/" % version
+        self.config_file_path = config_file_path
+        self.api_key = os.environ["DELIGHTED_API_KEY"]
+        self.endpoints = yaml.load(open(self.config_file_path), Loader=yaml.FullLoader).get('endpoints')
+        self.schema_prefix = yaml.load(open(self.config_file_path), Loader=yaml.FullLoader).get("schema_prefix")
+
+    def get_endpoint_data(self, endpoint, params):
+        url = self.url + endpoint + '.json'
+        return requests.get(url, params=params, auth=(self.api_key, ""))
+
+    def get_date_field(self, endpoint):
+        if not endpoint.get('date_fields'):
+            return ("created_at", "updated_at")
+        return endpoint.get('date_fields')
+
+    def get_table(self, endpoint_key):
+        endpoint = self.endpoints.get(endpoint_key)
+        if not endpoint.get('table'):
+            return endpoint_key + 's'
+        return endpoint.get('table')
+
+    def main(self, endpoint_key, batch_size=100, page=1, since=None, until=None):
+        params = {"page": page,
+                  "per_page": batch_size,
+                  "updated_since": since,
```

Comment:
en plus du commentaire ci dessous sur le cleaning, il faut qu'on parle de la recherche du since until directement intégré dans pyred/pyzure / dbstream ?